src/main/java/io/appform/dropwizard/actors/base/UnmanagedConsumer.java

```diff
@@ -63,77 +55,24 @@ public UnmanagedConsumer(
-        private String tag;
-
-        private Handler(Channel channel,
-                        ObjectMapper mapper,
-                        Class<? extends Message> clazz,
-                        int prefetchCount) throws Exception {
-            super(channel);
-            this.mapper = mapper;
-            this.clazz = clazz;
-            getChannel().basicQos(prefetchCount);
-        }
-
-        @Override
-        public void handleDelivery(String consumerTag, Envelope envelope,
-                                   AMQP.BasicProperties properties, byte[] body) throws IOException {
-            try {
-                final Message message = mapper.readValue(body, clazz);
-                boolean success = retryStrategy.execute(() -> handle(message, messageProperties(envelope)));
-                if (success) {
-                    getChannel().basicAck(envelope.getDeliveryTag(), false);
-                } else {
-                    getChannel().basicReject(envelope.getDeliveryTag(), false);
-                }
-            } catch (Throwable t) {
-                log.error("Error processing message...", t);
-                if (errorCheckFunction.apply(t)) {
-                    log.warn("Acked message due to exception: ", t);
-                    getChannel().basicAck(envelope.getDeliveryTag(), false);
-                } else if (exceptionHandler.handle()) {
-                    log.warn("Acked message due to exception handling strategy: ", t);
-                    getChannel().basicAck(envelope.getDeliveryTag(), false);
-                } else {
-                    getChannel().basicReject(envelope.getDeliveryTag(), false);
-                }
-            }
-        }
-
-        private MessageMetadata messageProperties(final Envelope envelope) {
-            return new MessageMetadata(envelope.isRedeliver());
-        }
-    }
-
     public void start() throws Exception {
         for (int i = 1; i <= config.getConcurrency(); i++) {
             Channel consumeChannel = connection.newChannel();
-            final Handler handler = new Handler(consumeChannel, mapper, clazz, prefetchCount);
+            final Handler<Message> handler = new Handler<>(consumeChannel, mapper, clazz,
+                    prefetchCount, errorCheckFunction, retryStrategy, exceptionHandler, handlerFunction);
             final String tag = consumeChannel.basicConsume(queueName, false, handler);
             handler.setTag(tag);
             handlers.add(handler);
             log.info("Started consumer {} of type {}", i, name);
         }
     }
 
-    public void stop() throws Exception {
+    public void stop() {
         handlers.forEach(handler -> {
             try {
                 final Channel channel = handler.getChannel();
                 channel.basicCancel(handler.getTag());
-                channel.close();
-                log.info("Consumer channel {} closed.", name);
+                log.info("Consumer channel {} cancelled.", name);
```

Comment:
Looks like next MR changes crept in here. Let's do this as part of next MR

Reply:
reverted.